components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -850,22 +865,24 @@ class GeckoEngineSession(
         }
 
         override fun onTitleChange(session: GeckoSession, title: String?) {
-            if (!privateMode) {
-                currentUrl?.let { url ->
-                    settings.historyTrackingDelegate?.let { delegate ->
-                        if (delegate.shouldStoreUri(url)) {
-                            // NB: There's no guarantee that the title change will be processed by the
-                            // delegate before the session is closed (and the corresponding coroutine
-                            // job is cancelled). Observers will always be notified of the title
-                            // change though.
-                            launch(coroutineContext) {
-                                delegate.onTitleChanged(url, title ?: "")
+            if (!isLastAppredirectedUri) {
```

Comment:
These two if checks can be combined on the same line.

Reply:
Actually I wanted the whole onTitleChange block to be executed including 
https://github.com/indurs/android-components/blob/232fc9c8f7de8b3eaa40a88441990a34828f150e/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt#L884
only if the isLastAppredirectedUri is false.

Reply:
Yes, I meant couldn't we do `if (!isLastAppRedirectedUri && !privateMode)`?

Reply:
I see, we stil lwant to notify the observers even if it's in private mode.  Got it thanks.